tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -106,1020 +100,834 @@ describe(__filename, () => {
+        userId: defaultUserId,
       }),
     );
   });
 
   it('fetches reviews if user is loaded', () => {
-    const { params, store } = signInUserWithUserId(100);
-    const user = getCurrentUser(store.getState().users);
+    signInUserWithProps();
 
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-    const errorHandler = createStubErrorHandler();
+    const dispatch = jest.spyOn(store, 'dispatch');
 
-    const page = 123;
-    const location = createFakeLocation({ query: { page } });
+    const page = '123';
+    const location = getLocation({ search: `?page=${page}` });
 
-    renderUserProfile({ errorHandler, location, params, store });
+    renderUserProfile({ location });
 
-    sinon.assert.calledWith(
-      dispatchSpy,
+    expect(dispatch).toHaveBeenCalledWith(
       fetchUserReviews({
-        errorHandlerId: errorHandler.id,
+        errorHandlerId: createErrorHandlerId(),
         page,
-        userId: user.id,
+        userId: defaultUserId,
       }),
     );
   });
 
   it('does not fetch reviews if already loaded', () => {
-    const { params, store } = signInUserWithUserId(100);
-    const user = getCurrentUser(store.getState().users);
+    signInUserWithProps();
 
-    _setUserReviews({ store, userId: user.id });
+    _setUserReviews();
 
-    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const dispatch = jest.spyOn(store, 'dispatch');
 
-    renderUserProfile({ params, store });
+    renderUserProfile();
 
-    sinon.assert.neverCalledWithMatch(dispatchSpy, {
-      type: FETCH_REVIEWS,
-    });
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ 'type': FETCH_USER_REVIEWS }),
+    );
   });
 
   it(`displays the user's reviews`, () => {
-    const userId = 100;
-    const { params, store } = signInUserWithUserId(userId);
+    signInUserWithProps();
 
     const review = fakeReview;
     const reviews = [review];
-    _setUserReviews({ store, userId, reviews });
-    const location = createFakeLocation({ query: { foo: 'bar' } });
-
-    const root = renderUserProfile({ location, params, store });
+    _setUserReviews({ reviews });
 
-    expect(root.find('.UserProfile-reviews')).toHaveLength(1);
-    expect(root.find('.UserProfile-reviews')).toHaveProp(
-      'header',
-      'My reviews',
-    );
-    expect(root.find('.UserProfile-reviews')).toHaveProp('footer', null);
+    renderUserProfile();
 
-    expect(root.find(AddonReviewCard)).toHaveProp(
-      'review',
-      createInternalReview(review),
-    );
+    expect(screen.getByText('My reviews')).toBeInTheDocument();
+    expect(screen.getByText(fakeReview.body)).toBeInTheDocument();
   });
 
   it(`displays the user's reviews with pagination when there are more reviews than the default API page size`, () => {
-    const userId = 100;
-    const { params, store } = signInUserWithUserId(userId);
+    signInUserWithProps();
 
     const reviews = Array(DEFAULT_API_PAGE_SIZE).fill(fakeReview);
```

Comment:
I've hit a problem with this test - the line `expect(screen.queryAllByText('posted')).toHaveLength(DEFAULT_API_PAGE_SIZE);` below is failing - the length is actually the length of `reviews` (i.e. it's 27, rather than 25).

Reply:
That's because I was wrong. :-(  I *thought* it made sense to fill the array with `DEFAULT_API_PAGE_SIZE + 2` reviews, but that's not actually what would happen in the real world. The array would contain all of the reviews returned by the API, and, because of paging, we would only get `DEFAULT_API_PAGE_SIZE` reviews returned. It was actually fine as it was before. My apologies for this red herring. 😢